src/detail/header.cpp

```diff
@@ -524,6 +525,11 @@ on_insert_content_length(
         // one value
         md.content_length.ec = {};
         md.content_length.value = *rv;
+        if( md.expect.ec == error::expect_needs_content )
```

Comment:
hmm no this is not ok. You can't signal an error and then take it back later. You can only set the error when you know for certain that an error exists. Otherwise users are going to have a hard time if they, for example, want to know "if a terminal error has occurred yet."